packages/flutter/lib/src/widgets/dismissible.dart

```diff
@@ -30,6 +30,12 @@ typedef DismissDirectionCallback = void Function(DismissDirection direction);
 /// Used by [Dismissible.confirmDismiss].
 typedef ConfirmDismissCallback = Future<bool?> Function(DismissDirection direction);
 
+/// Signature used by [Dismissible] to give the application an opportunity to
+/// accept or deny a dismiss gesture.
+///
+/// Used by [Dismissible.shouldDismiss].
+typedef AcceptDismissCallback = bool? Function(AcceptDismissDetails details);
```

Comment:
I like the choice to make this nullable—a fallback to default behavior could make the implementation much more simple :)